dask/array/core.py

```diff
@@ -1047,6 +1047,28 @@ def _set_chunks(self, chunks):
     def __len__(self):
         return sum(self.chunks[0])
 
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        out = kwargs.get('out', ())
+        for x in inputs + out:
+            if not isinstance(x, (np.ndarray, Number, Array)):
```

Comment:
What's the desired behavior when interacting with, say xarray.DataArrays? I think let them decide how to handle the inputs (so NotImplemented is correct here).